openapi/components/schemas/QuickBooksOnlineExternalServiceSettings.yaml

```
@@ -0,0 +1,123 @@
+
+type: object
+description: QuickBooks Online settings.
+properties:
+  customer:
+    type: object
+    title: QuickBooks Online customer external service settings
+    properties:
+      sync:
+        type: string
+        description: Specifies how to sync customers.
+        default: manually
+        enum:
+          - manually
+          - when-used
+          - always
+        x-enumDescriptions:
+          manually: Only via a manual sync.
+          when-used: Whenever related entity is synced (transaction, invoice).
+          always: Create and update customers automatically.
+      displayName:
+        type: string
+        default: full-name
+        enum:
+          - id
+          - full-name
+          - organization-name
+  invoice:
+    type: object
+    title: QuickBooks Online invoice external service settings
+    properties:
+      sync:
+        type: string
+        description: Specifies how to sync invoices.
+        default: manually
+        enum:
+          - manually
+          - only-create
+          - always
+        x-enumDescriptions:
+          manually: Only via a manual sync.
+          only-create: Only create invoices on issue without updating.
+          always: Create, update and void invoices automatically.
```

nitpick: Good to give more detail

```suggestion
          manually: Customer data is only synchronized when it is manually executed.
          when-used: Customer data is synchronized when related entities, such as transactions or invoices are changes or updated.
          always: Customer data is synchronized automatically as data is created or updated.
```

---

This is not accurate related to invoices. It's more like `when-issued`. Adjusted to be that.

```suggestion
          manually: Invoice data is only synchronized when it is manually executed.
          when-issued: Invoice data is synchronized when issued.
          always: Invoice data is synchronized automatically as data it is issued or updated.
```